api/src/main/java/com/exadel/frs/core/trainservice/controller/FaceController.java

```diff
@@ -16,71 +16,110 @@
+            @ApiParam(value = "Specify whether the model should start retraining immediately after the request is completed " +
+                    "(set this parameter to value \"no\", if operating with a lot of images one after another). " +
+                    "Allowed values: \"yes\", \"no\", \"force\". \"Force\" option will abort already running processes of " +
+                    "classifier training.")
+            @RequestParam(value = "retrain", required = false, defaultValue = "force") final String retrainOption,
+            @ApiParam(value = "The minimal percent confidence that found face is actually a face.")
+            @RequestParam(value = "det_prob_threshold", required = false) final Double detProbThreshold,
+            @ApiParam(value = "api key", required = true)
+            @RequestHeader(X_FRS_API_KEY_HEADER) final String apiKey
+    ) throws IOException {
+
+        imageValidator.validate(file);
+        Face face = scanService.scanAndSaveFace(file, faceName, detProbThreshold, apiKey);
+        getTrainingOption(retrainOption).run(apiKey, retrainService);
+        return faceMapper.toResponseDto(face);
+    }
 
     @GetMapping
     public Map<String, List<FaceResponseDto>> findFacesByModel(
-            @RequestHeader(name = X_FRS_API_KEY_HEADER)
-            final String apiKey
+            @RequestHeader(name = X_FRS_API_KEY_HEADER) final String apiKey
     ) {
         return Map.of("faces", faceMapper.toResponseDto(faceService.findFaces(apiKey)));
     }
 
     @WriteEndpoint
-    @DeleteMapping("/name/{faceName}")
-    public void deleteFaceByName(
-            @PathVariable
-            final String faceName,
-            @RequestHeader(name = X_FRS_API_KEY_HEADER)
-            final String apiKey,
-            @RequestParam(name = "retrain", required = false)
-            final String retrain
-    ) {
-        faceService.deleteFaceByName(faceName, apiKey, retrain);
-    }
-
-    @WriteEndpoint
-    @DeleteMapping("/{id}")
-    public void deleteFaceById(
-            @PathVariable
-            final String id,
-            @RequestHeader(name = X_FRS_API_KEY_HEADER)
-            final String apiKey,
-            @RequestParam(name = "retrain", required = false)
-            final String retrain
+    @DeleteMapping
+    public List<FaceResponseDto> deleteFaces(
+            @RequestParam(name = "subject", required = false) final String subject,
+            @RequestParam(value = "retrain", required = false, defaultValue = "force") final String retrain,
+            @RequestHeader(name = X_FRS_API_KEY_HEADER) final String apiKey
     ) {
-        faceService.deleteFaceById(id, apiKey, retrain);
+        val faces = new ArrayList<Face>();
+        if (!isEmpty(subject)) {
+            faces.addAll(faceService.deleteFaceByName(subject, apiKey));
+            if (!faces.isEmpty()) {
+                getTrainingOption(retrain).run(apiKey, retrainService);
```

Comment:
we need to check how many faces are left in the model before retraining. In the case of one face in a model after deletion, we will have the Exception of retraining.